Content.Server/_FarHorizons/Medical/Surgery/SurgeryOverhaulSystems.cs

```diff
@@ -0,0 +1,185 @@
+        {
+            if (surgProto.TryGetComponent<SurgeryTechnologyComponent> (out var techvar, _componentFactory) && TryComp(args.Body, out BuckleComponent? buckle)
+                && TryComp(buckle.BuckledTo, out DeviceLinkSinkComponent? linkComp) && linkComp.LinkedSources.Count > 0 &&
+                TryComp<TechnologyDatabaseComponent>(linkComp.LinkedSources.First(), out var techComp))
+            {
+                foreach (var (key, value) in techvar.TechnologyModifier!)
+                {
+                    var TechProto = _prototypes.Index<TechnologyPrototype>(key.Id);
+                    if (_research.IsTechnologyUnlocked(uid, TechProto, techComp) && ResearchModifier > value)
+                        ResearchModifier = value;
+                }
+            }
+
+            if (TryComp<DamageableComponent>(args.Body, out var dmgComp))
+                foreach (var key in healComp.Damage!.DamageDict.Keys)
+                    BonusHeal.DamageDict.Add(key, dmgComp.TotalDamage / ResearchModifier);
+
+            TotalHeal = healComp.Damage! + (-BonusHeal);
+            _damageableSystem.TryChangeDamage(args.Body, TotalHeal);
+        }
+    }
+    private void OnNecrosisComplete(EntityUid uid, NecrosisSurgeryComponent comp, ref SurgeryStepCompleteEvent args)
+    {
+        if (_net.IsClient) return;
+        LoadSurgeriesForRotten();
+
+        if (!TryComp<NecrosisSurgeryComponent>(args.Part, out var surgComp))
+            return;
+
+        var surgProto = _prototypes.Index<EntityPrototype>(args.SurgeryProto);
+        if (!surgProto.TryGetComponent<NecrosisSurgeryComponent>(out var surgProtoComp, _componentFactory)) return;
+        surgComp.RequiredSurgeries.Clear();
+
+        for (int i = 0; i < surgProtoComp.AmountOfSurgeries; i++)
+        {
+            EntProtoId chosenSurgery;
+            if (_surgeriesForRotten.Count == 0)
+                break;
+
+            while (true)
+            {
+                chosenSurgery = _random.PickAndTake(_surgeriesForRotten);
+                if (!_entity.TryGetSingleton(chosenSurgery, out var surgeryEnt))
+                    continue;
+                var ev = new SurgeryValidEvent(args.Body, args.Part);
+
+                RaiseLocalEvent(surgeryEnt, ref ev);
+
+                if (!ev.Cancelled)
+                    break;
+            }
+            surgComp.RequiredSurgeries.Add(chosenSurgery!);
+        }
+    }
+    private void OnRepairEyesComplete(EntityUid uid, SurgeryRepairEyesComponent comp, ref SurgeryStepCompleteEvent args)
+    {
+        if (TryComp<BlindableComponent>(args.Body, out var blindComp))
+            _blindableSystem.AdjustEyeDamage(args.Body, -blindComp.EyeDamage);
+    }
+    // Valid Event Checks
+    private void OnNecrosisSurgeryValid(Entity<NecrosisSurgeryComponent> ent, ref SurgeryValidEvent args)
+    {
+        if (!_rottingSystem.IsRotten(args.Body))
```

Comment:
```suggestion
    {
        if (args.Cancelled)
            return;
    
        if (!_rottingSystem.IsRotten(args.Body))
```

Good practice to not run your code when not necessary.

Reply:
Added